### DISCUSSION TITLE
init config does not produce correct config for transformer, ner pipeline

### DISCUSSION THREAD

**Discussion by salman1993** [explosion/spaCy#7023] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour
If you run:
`spacy init config --lang en --pipeline transformer,ner test.cfg`

The config produced contains `tok2vec` instead of the desired `transformer`  

## Your Environment
- **spaCy version:** 3.0.1
- **Platform:** Darwin-19.6.0-x86_64-i386-64bit
- **Python version:** 3.7.9

**Comment by DeltaSierra4** [comment, marked as answer]
If you add the -G option for GPU usage, such as:
`spacy init config --lang en --pipeline transformer,ner -G test.cfg`
Then the resulting configuration file contains components.transformer

**Comment by svlandeg** [comment]
That's correct; @DeltaSierra4. The "pipeline" argument is not actually supposed to include `tok2vec` or `transformer`. Instead, the type of embedding component is decided based on the GPU/CPU choice and efficiency/accuracy.

If you feel this is unclear from the docs - suggestions on how to improve would be appreciated as a PR!

**Comment by salman1993** [comment]
Sorry for the confusion and thank you!
